tests/unit/storage/test_gcp.py

```diff
@@ -1,16 +1,73 @@
 import gzip
-import io
+import logging
 import tempfile
+from pathlib import Path
 from unittest.mock import MagicMock, patch
 
 import pytest
+import vcr
+import zstandard
 from google.cloud import storage as google_storage
 from google.resumable_media.common import DataCorruption
 
 from shared.storage.exceptions import BucketAlreadyExistsError, FileNotInStorageError
 from shared.storage.gcp import GCPStorageService
 from tests.base import BaseTestCase
 
+
+def before_record_cb(request):
+    if request.uri == "https://oauth2.googleapis.com/token":
+        request.body = b""
+
+    if request.body is not None:
+        try:
+            body = request.body
+            body = body.splitlines()
+            # there's random data in lines that start with the pattern below
+            body = [line for line in body if not line.startswith(b"--===============")]
+            body = b"\n".join(body)
+            request.body = body
+        except Exception as e:
+            raise e
+
+        # there's also random data in gzip compressed data so we decompress it
+        if (start := request.body.find(b"\x1f\x8b\x08\x00")) > 0:
+            compressed_body = request.body[start:]
+            request.body = request.body[:start] + gzip.decompress(compressed_body)
```

Comment:
there shouldn’t be any "random data" here. I would rather not mess with these fixtures, as I have little confidence in these to begin with :-D

Reply:
the random data is the timestamp in in the gzip compression, afaik there's no way to avoid having it there, but basically it causes the VCR matching for requests to fail because the body of the request _is_ different

i think this is necessary if we want to ensure that the body of the requests is correct, which is how we ensure that the compression is occuring properly